### DISCUSSION TITLE
use-form suggestion: handleChange hook

### DISCUSSION THREAD

**Discussion by cstrat** [mantinedev/mantine#61] [Q&A] [answered]
Happy to put this in as a PR, but want to check if you think it is in-line with the plan for the hook.
I typically have this boilerplate in any form to handle form field changes.
It would work great in the `use-form` hook.

````
const handleChange = (event) => {
    const { id, type, checked } = event.target;
    let { name, value } = event.target;

    if (type === "checkbox") {
        value = checked;
    }

    if (id) {
        name = id;
    }

    // Whatever the following needs to be to set the state.
    form.setFieldValue(name, value);
};
````

This lets you simply attach the `handleChange` function to the onChange event of a form.
In the above it will use an `id` over the `name` but can be whatever you prefer.

Gets rid of having to do things like: `onChange={(event) => form.setFieldValue('name', event.currentTarget.value)}` where it is easy to mis-match the field name etc...
You would just have: `onChange={form.handleChange}` - but could still use the former so non-breaking change to add.

**Comment by rtivital** [comment, marked as answer]
I've tried to implement something like this and it did not work out:
- some inputs do not support event (NumberInput, RadioGroup, upcoming [SegmentedControl](https://github.com/mantinedev/mantine/blob/next-1.1.0/src/mantine-core/src/SegmentedControl/SegmentedControl.tsx)) – it won't work for them and experience will be inconsistent 
- TypeScript – this will bypass all type checks – there is no way to validate that event name or id is presented in use-form state, thus although form type can be maintained it will result in possibilities of types mismatches (e.g. incorrect field name, string instead of boolean in values) making form type obsolete.

**Reply by cstrat** [reply]
Yeah thought there might be a reason!
I don't understand the typescript reasoning (haven't tried using it myself yet), but I get what you mean about inconsistencies with other form field types. I actually didn't realise it wouldn't work on Number or Radio (rarely use those form fields - that's probably why!)

I guess I will keep including my boilerplate then 👍🏻
